### ISSUE TITLE
Better Quest Script Organization

### ISSUE THREAD

**Issue by Tocharaeh** [4dmud/mud#26]
(this is a copy and paste from my discord post)

I've been ruminating on quest cards. The amount of work that is involved is obvious, so I will avoid being redundant. What I have come to the conclusion of is that we need to set a new standard for quest script nomenclature, and have the quest vars listed in order for easy acquisition when doing a tlist of a given zone.

I've an idea on this to streamline: In the trigger edit menu, we need an option to set the quest nomenclature: Here is my idea:

TRIGEDIT MENU
1) Name             : MICROFILAMENT 8316 - DMG if worn on face
2) Quest Nom    : **ExampleQuest_1 (matching the step in the quest var)**
3) Intended for   : Objects
4) Trigger types: Random 
5) Numeric Arg  : 30 (chance of firing)
6) Arguments    : 
7) Commands
Q) Quit

TLIST OF ZONE 201
Index VNum    Trigger Name                        Type
----- ------- -------------------------------------------------------
   1) [20100] Beo Tradepoint Shop - buy listed items        mob Command 
   2) [20101] Gelindril Army time LOAD at 1800                  wld Global Time 
   3) [20102] Gelindril Army time PURGE                            mob Global Time 
   4) [20103] Hume Glassmend - Register Trigger              mob Receive 
   5) [20104] Jorgin's Hunter Quest Script - Step 3 - 1 of   mob Speech 
   6) [20105] Jorgin's Hunter - Multiple Receive                 mob Receive **<ExampleQuest_1>**
   
   (picture example)
   
![image](https://user-images.githubusercontent.com/51241466/140159550-a7f5605d-4918-42ec-8d16-538a86e17010.png)

**Comment by loran4d** [comment]
I think adding a different view option for -q to display the alternate view, similar to how the Who command has around 5 filter/display options

display so tlist -q <zonenum/range>
current:
6) [20105] Jorgin's Hunter - Multiple Receive            mob Receive
alt -q
6) [20105] Jorgin's Hunter - Multiple Receive            <ExampleQuest_1>

**Comment by loran4d** [comment]
Wip update to trigedit menu, and tlist with the new quest nom col
still working on the spacing for the column headers

[issue26_quest_nom.patch](https://github.com/user-attachments/files/29316865/issue26_quest_nom.patch)

Option B added for Quest Nom, in dg_scripts.h, dg_olc.h, dg_olc.cpp, dg_db_scripts.cpp, dg_handler.cpp
Trigger Editor [20105]

1) Name         : Jorgin's Hunter - Multiple Receive
B) Quest Nom    : JorginHunterQuest_1
2) Intended for : Mobiles
3) Trigger types: Receive
4) Numeric Arg  : 0
5) Arguments    : <none>
6) Commands
Q) Quit

tlist output

<img width="1062" height="418" alt="Image" src="https://github.com/user-attachments/assets/1fda24de-d3d5-49f7-923b-df2bb6204fd2" />

**Comment by Rynald4d** [comment]
Good idea. It'd also be nice if the one who makes the questcard puts a comment in one of the triggers that uses a questflag to explain what the flag is used for. That would help other imms a lot later on.
I'd leave the (Issue #46) out of the code though, you can put #46 at the end of the commit message to link the patch to the issue.
